src/components/UniqueSlug/index.tsx

```diff
@@ -18,25 +18,39 @@ export const UniqueSlug: React.FC<{
   teamID?: string
   label?: string
   docID?: string
-}> = ({ initialValue, collection = 'teams', path = 'slug', label = 'Slug', teamID, docID }) => {
-  const [value, setValue] = React.useState(initialValue)
-  const debouncedValue = useDebounce(value, 100)
+  validateOnInit?: boolean
+}> = ({
+  initialValue,
+  collection = 'teams',
+  path = 'slug',
+  label = 'Slug',
+  teamID,
+  docID,
+  validateOnInit = true,
```

Comment:
I don't think we should initialize this as `true`. It should be the other way around, so that components have to  _opt-in_ with `validateOnInit`